kolibri/core/analytics/utils.py

```diff
@@ -169,7 +177,19 @@ def extract_channel_statistics(channel):
     }
 
 
-def cmp_semver(kolibri_version, version_range):
-    if version_range == '*':
-        return True
-    return semver.match(kolibri_version, version_range)
+@transaction.atomic
+def create_and_update_notifications(data, source):
+    if data.get('messages'):
```

Comment:
This means that if we don't receive any messages, we won't do anything. But in that case we'd want to mark all messages as inactive. So, probably best to just do `messages = data.get('messages', [])` and continue from there.